packages/devtools_app/lib/src/memory/flutter/memory_controller.dart

```diff
@@ -546,47 +895,118 @@ class MemoryTimeline {
   ValueNotifier<bool> get pausedNotifier => _pausedNotifier;
 
   /// dart_ui.Image Image asset displayed for each entry plotted in a chart.
+  // ignore: unused_field
   dart_ui.Image _img;
 
+  // TODO(terry): Look at using _img for each data point (at least last N).
+  dart_ui.Image get dataPointImage => null;
+
   set image(dart_ui.Image img) {
     _img = img;
   }
 
+  void reset() {
+    liveData.clear();
+    startingIndex = 0;
+    chartData.reset();
+    engineChartData.reset();
+  }
+
   /// Common utility function to handle loading of the data into the
   /// chart for either offline or live Feed.
-  List<Map> _processData(List<HeapSample> data, int startingDataIndex) {
+  ///
+  /// [startingDataIndex] if != -1 then use this parameter to process the newest
+  /// samples not yet plotted.
+  /// [lastMinutes] if != -1, then use this parameter to plot last n ninutes of data.
+  ///
+  /// [ArgumentError] if thrown requires either startingDataIndex or lastMinutes
+  /// to be passsed.
+  List<Map> _processData(int index) {
     final result = <Map<String, Entry>>[];
 
-    for (var dataIndex = startingDataIndex;
-        dataIndex < data.length;
-        dataIndex++) {
-      final sample = data[dataIndex];
+    var lastIndex = index;
+    for (; lastIndex < data.length; lastIndex++) {
```

Comment:
put `var lastIndex = index` in loop declaration

Reply:
ALready done - thx